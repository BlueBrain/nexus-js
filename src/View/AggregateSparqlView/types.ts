import { SparqlViewResponse } from '../SparqlView/types';

export interface AggregateViewReference {
  project: string;
  viewId: string;
}
export interface AggregateSparqlViewResponse extends SparqlViewResponse {
  views: AggregateViewReference[];
}
