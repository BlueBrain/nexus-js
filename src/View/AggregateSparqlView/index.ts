import { AggregateSparqlViewResponse, AggregateViewReference } from './types';
import SparqlView from '../SparqlView';

export default class AggregateSparqlView extends SparqlView {
  views: AggregateViewReference[];
  constructor(
    orgLabel: string,
    projectLabel: string,
    aggregateSparqlViewResponse: AggregateSparqlViewResponse,
  ) {
    super(orgLabel, projectLabel, aggregateSparqlViewResponse);
    this.views = aggregateSparqlViewResponse.views;
  }
}
