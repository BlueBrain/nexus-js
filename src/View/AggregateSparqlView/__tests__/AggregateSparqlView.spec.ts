import AggregateSparqlView from '../';

const mockAggregatedSparqlViewGetResponse = {
  '@context': [
    'https://bluebrain.github.io/nexus/contexts/view.json',
    'https://bluebrain.github.io/nexus/contexts/resource.json',
  ],
  '@id': 'http://example.com/base/aggregated',
  '@type': ['View', 'AggregateSparqlView'],
  _uuid: 'c166385c-c2c3-4815-b82e-b4ba5bef6910',
  views: [
    {
      project: 'my-org/my-project',
      viewId: 'http://example.com/base/custom',
    },
    {
      project: 'my-org/my-project-2',
      viewId: 'http://example.com/base/custom',
    },
  ],
  _self: 'http://example.com/v1/views/my-org/my-project/aggregated',
  _constrainedBy: 'https://bluebrain.github.io/nexus/schemas/view.json',
  _project: 'http://example.com/v1/projects/my-org/my-project',
  _rev: 1,
  _deprecated: false,
  _createdAt: '2019-04-10T08:58:39.530Z',
  _createdBy: 'http://example.com/v1/realms/github/users/somebody',
  _updatedAt: '2019-04-10T08:58:39.530Z',
  _updatedBy: 'http://example.com/v1/realms/github/users/somebody',
};

describe('AggregateSparqlView', () => {
  describe('aggregateSparqlView', () => {
    it('should have all the properties created by the response', () => {
      const aggregateSparqlView = new AggregateSparqlView(
        'myOrg',
        'myProject',
        mockAggregatedSparqlViewGetResponse,
      );
      expect(aggregateSparqlView).toMatchSnapshot();
    });
  });
});
