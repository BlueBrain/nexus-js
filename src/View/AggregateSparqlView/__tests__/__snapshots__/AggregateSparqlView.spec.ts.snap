// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregateSparqlView aggregateSparqlView should have all the properties created by the response 1`] = `
AggregateSparqlView {
  "deprecated": false,
  "id": "http://example.com/base/aggregated",
  "orgLabel": "myOrg",
  "projectLabel": "myProject",
  "queryURL": "/views/myOrg/myProject/http%3A%2F%2Fexample.com%2Fbase%2Faggregated/sparql",
  "rev": 1,
  "type": Array [
    "View",
    "AggregateSparqlView",
  ],
  "uuid": "c166385c-c2c3-4815-b82e-b4ba5bef6910",
  "views": Array [
    Object {
      "project": "my-org/my-project",
      "viewId": "http://example.com/base/custom",
    },
    Object {
      "project": "my-org/my-project-2",
      "viewId": "http://example.com/base/custom",
    },
  ],
}
`;
